iam_role.go

```diff
@@ -220,6 +249,121 @@ func (c *Client) CreateIamTrustRole(roleName string, roleType string, trustArn s
 	return cr, nil
 }
 
+type IamRoleInput struct {
+	RoleName                    *string            `json:"roleName"`
+	RoleType                    *string            `json:"roleType,omitempty"`
+	IncDefPols                  *int               `json:"includeDefaultPolicy,omitempty"`
+	AlksAccess                  *bool              `json:"enableAlksAccess,omitempty"`
+	TrustArn                    *string            `json:"trustArn,omitempty"`
+	TemplateFields              *map[string]string `json:"templateFields,omitempty"`
+	MaxSessionDurationInSeconds *int               `json:"maxSessionDurationInSeconds,omitempty"`
+	Tags                        *[]Tag             `json:"tags,omitempty"`
+}
+
+type IamRoleOutput struct {
+	BaseResponse
+	RoleName                    *string            `json:"roleName"`
+	RoleType                    *string            `json:"roleType,omitempty"`
+	RoleArn                     *string            `json:"roleArn,omitempty"`
+	RoleIPArn                   *string            `json:"instanceProfileArn,omitempty"`
+	RoleAddedToIP               *bool              `json:"addedRoleToInstanceProfile,omitempty"`
+	Exists                      *bool              `json:"roleExists,omitempty"`
+	TemplateFields              *map[string]string `json:"templateFields,omitempty"`
+	MaxSessionDurationInSeconds *int               `json:"maxSessionDurationInSeconds,omitempty"`
+	Tags                        *[]Tag             `json:"tags,omitempty"`
+}
+
+type encodeDetails struct {
+	*AccountDetails
+	*IamRoleInput
+}
+
+type requestOp struct {
+	ReqObj   []byte
+	Method   string
+	Endpoint string
+	Response *IamRoleOutput
+}
+
+func (enc *encodeDetails) MarshalJSON() ([]byte, error) {
+	obj, e := json.Marshal(*enc)
+	if e != nil {
+		return nil, e
+	}
+	return obj, e
+}
+
+func (c *Client) makeIamRoleRequest(operation *requestOp) error {
```

Comment:
This function seems like it is explicitly for update role, so maybe change the name to reflect that, or have the error message dynamically print the endpoint that it's parsing a response for.

Reply:
The goal is to use this universally. It's not intrusive to any other functions at present, but it can easily abstract the request/response calls from all individual APIs